### ISSUE TITLE
Microsoft.Extensions.Options ValidateOptionsResult when success

### ISSUE THREAD

**Issue by SinxHe** [aspnet/Home#3594]
I've found a feature in `Microsoft.Extensions.Options.DataAnnotations` when i want to develop a similarity.

When I view the source code. I notice this below in the `ValidateOptionsResult`:

```csharp
/// <summary>
/// Validation was successful.
/// </summary>
public static readonly ValidateOptionsResult Success = new ValidateOptionsResult() 
{ 
    Skipped = true 
};
```

I am confused why successful result set `Skipped` property true rather than `Succeeded`. Is it a design or some other reasons?

On the other hand, in the source code of `DataAnnotationValidateOptions`, the annotation of code shows it will validate all options and it is a useful feature, but the code of `if (Name == null || name == Name)` appears will not validate all options when the options have not null name and i develivery null to `Validate` method:

```csharp
/// <summary>
/// Validates a specific named options instance (or all when name is null).
/// </summary>
/// <param name="name">The name of the options instance being validated.</param>
/// <param name="options">The options instance.</param>
/// <returns>The <see cref="ValidateOptionsResult"/> result.</returns>
public ValidateOptionsResult Validate(string name, TOptions options)
{
    // Null name is used to configure all named options.
    if (Name == null || name == Name)
    {
        var validationResults = new List<ValidationResult>();
        if (Validator.TryValidateObject(options,
            new ValidationContext(options, serviceProvider: null, items: null), 
            validationResults, 
            validateAllProperties: true))
    ...
    // Ignored if not validating this instance.
    return ValidateOptionsResult.Skip;
}
```

**Comment by SinxHe** [comment]
@HaoK

**Comment by HaoK** [comment]
Thanks yeah that's just a typo, I'll fix it tomorrow

**Comment by HaoK** [comment]
fyi @ajcvickers new 2.2 bug...  Got bitten by options issues not showing up due to no longer being in the Options repo that I watch since this was filed 15 days ago

**Comment by HaoK** [comment]
> he code of if (Name == null || name == Name) appears will not validate all options when the options have not null name 

Yes this is just how named options work in general for Configure/Validate, neither is applied if the action isn't targetting the specific name (null is a special value used to mean it will target all options instances)

**Comment by SinxHe** [comment]
Thank you for your reply. Look forward to your repair.

**Comment by ajcvickers** [comment]
@HaoK Thanks. Presumably the fix is low risk for 2.2?

**Comment by HaoK** [comment]
Yeah should be a one liner + a test or two verifying that we set success properly, the code today was only looking for failures so there wasn't really a difference between Success or Skipped today
